tests/spread/plugins/v2/build-and-run-hello/task.yaml

```diff
@@ -0,0 +1,46 @@
+summary: Build, rebuild and run hello.c with different plugin configurations
+
+environment:
+  SNAP/autotools: autotools-hello
+  SNAP/cmake: cmake-hello
+  SNAP/make: make-hello
+  SNAP/python: python-hello
+
+prepare: |
+  #shellcheck source=tests/spread/tools/snapcraft-yaml.sh
+  . "$TOOLS_DIR/snapcraft-yaml.sh"
+  set_base "../snaps/$SNAP/snap/snapcraft.yaml"
+
+restore: |
+  cd "../snaps/$SNAP"
+  snapcraft clean
+  rm -f ./*.snap
+
+  # Undo changes to hello
```

Comment:
could we do a `git reset --hard HEAD` instead of the checkout and the restore_yaml?

Reply:
and maybe a `git clean -f -d -x -f` for the remainder?

Reply:
let's discuss this one after, I was just doing some spread optimizations (for faster uploads) with include and exclude that are incompatible with this.

Reply:
:+1: